library/src/main/java/bio/terra/pfb/PfbReader.java

```diff
@@ -0,0 +1,211 @@
+package bio.terra.pfb;
+
+import bio.terra.pfb.exceptions.InvalidPfbException;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.avro.Schema;
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.file.DataFileStream;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.specific.SpecificDatumReader;
+
+public class PfbReader {
+
+  public String showSchema(String fileLocation) {
+    try {
+      Schema schema = getSchema(fileLocation);
+      List<Schema> pfbSchemaShow =
+          schema.getField("object").schema().getTypes().stream()
+              .filter(t -> !t.getName().equals("Metadata"))
+              .toList();
+      String show = pfbSchemaShow.stream().map(Schema::toString).toList().toString();
+      return convertEnum(show);
+    } catch (IOException e) {
+      return getErrorMessage(e);
```

Comment:
No change needed here, but is it important for this library _not_ to throw exceptions?  My instinct would be to let these exceptions bubble up assuming they are truly abnormal behavior, and let the CLI layer decide whether or not to convert them to text output.  Wondering if you think this would be a reasonable change.

Reply:
It looks like picocli also supports `Callable` which does let you throw an exception.

Alternatively, you can catch checked exceptions and rethrow using an unchecked exception, such as one of the `PicocliException` subclasses.

Reply:
I went with the second option to catch the exception and rethrow as PicocliException.